pkg/api/helpers.go

```diff
@@ -154,6 +154,17 @@ func IsStandardQuotaResourceName(str string) bool {
 	return standardQuotaResources.Has(str)
 }
 
+// AddStandardQuotaResourceNames add the resource names into the list of
+// resources known to the quota tracking system
+func AddStandardQuotaResourceNames(names ...ResourceName) {
```

Comment:
I think the whole thing could be extensibly factored by moving the validation rules to the validation package, registering quantity validation based on resourcename (even for /api) during api install, and running the plugged validators in validation.

I'm ok with this an intermediate step and a followup issue.